### DISCUSSION TITLE
Trigger haptic feedback from worklet

### DISCUSSION THREAD

**Discussion by pistonsky** [software-mansion/react-native-reanimated#2787] [Q&A]
Hello and thanks for maintaining this awesome library!

I would like to trigger haptic feedback from within a worklet - when user scrolls my custom range selector. I'm currently doing that using `Animated` from react-native combined with a callback, and it works, but sometimes it starts to lag. I would like to use `Reanimated` here and execute `[feedbackGenerator selectionChanged]` from within a worklet. Is there a way to do that?

**Comment by Noitidart** [comment]
I think you'll have to create a js worklet like this:

function vibrate() {
  'worklet';
  VibrateApi.vibrate();
}

then from your reanimated worklets call this with `runOnJS(vibrate)(()`
